runtime/src/asset.rs

```diff
@@ -1197,6 +1206,96 @@ decl_module! {
             Self::deposit_event(RawEvent::IdentifiersUpdated(ticker, identifiers));
             Ok(())
         }
+
+        /// Whitelisting the Smart-Extension address for a given ticker
+        ///
+        /// # Arguments
+        /// * `origin` - Signer who owns to ticker/asset
+        /// * `ticker` - ticker for whom extension get added
+        /// * `extension_details` - Details of the smart extension
+        pub fn add_extension(origin, ticker: Ticker, extension_details: SmartExtension<T::AccountId>) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let sender_key = Key::try_from(sender.encode())?;
+            let my_did =  match <identity::Module<T>>::current_did() {
+                Some(x) => x,
+                None => {
+                    if let Some(did) = <identity::Module<T>>::get_identity(&sender_key) {
+                        did
+                    } else {
+                        return Err(Error::<T>::DIDNotFound.into());
+                    }
+                }
+            };
+            ticker.canonize();
+            ensure!(Self::is_owner(&ticker, my_did), Error::<T>::UnAuthorized);
+
+            // Verify the details of smart extension & store it
+            ensure!(!<ExtensionDetails<T>>::exists((ticker, &extension_details.extension_id)), Error::<T>::ExtensionAlreadyPresent);
+            <ExtensionDetails<T>>::insert((ticker, &extension_details.extension_id), extension_details.clone());
+            <Extensions<T>>::mutate((ticker, &extension_details.extension_type), |ids| {
+                ids.push(extension_details.extension_id.clone())
+            });
+            Self::deposit_event(RawEvent::ExtensionAdded(ticker, extension_details.extension_id, extension_details.extension_name, extension_details.extension_type));
+            Ok(())
+        }
+
+        /// Archived the extension. Extension will not be used to verify the compliance or any smart logic it posses
+        ///
+        /// # Arguments
+        /// * `origin` - Signer who owns the ticker/asset.
+        /// * `ticker` - Ticker symbol of the asset.
+        /// * `extension_id` - AccountId of the extension that need to be archived
+        pub fn archive_extension(origin, ticker: Ticker, extension_id: T::AccountId) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let sender_key = Key::try_from(sender.encode())?;
+            let my_did =  match <identity::Module<T>>::current_did() {
```

Comment:
Please call a function as described above instead of copy-pasting.

Reply:
same as above